tests/conftest.py

```diff
@@ -352,16 +352,20 @@ def _role_factory(*, permission):
 
         name = f"Role_{permission}"
         assert name.isidentifier(), f"{name} is not a valid Python identifier"
-        Role = type(
-            name,
-            (object,),
-            {
-                "__module__": jobserver.authorization.roles.__name__,
-                "models": [],
-                "permissions": [permission],
-            },
-        )
-        setattr(jobserver.authorization.roles, name, Role)
+
+        Role = getattr(jobserver.authorization.roles, name, None)
```

Comment:
Thank you for doing this. It's a nuance that I hadn't realised before.

Reply:
I can't remember when it popped into my head, but I think it's worth making it explicit.